### ISSUE TITLE
Bun build throws Reference Error when importing CJS code with `__esModule: true`

### ISSUE THREAD

**Issue by arthurfiorette** [oven-sh/bun#5654]
### What version of Bun is running?

1.0.2+37edd5a6e389265738e89265bcbdf2999cb81a49

### What platform is your computer?

Linux 5.15.90.1-microsoft-standard-WSL2 x86_64 x86_64

### What steps can reproduce the bug?

```js
// lib.js
Object.defineProperty(exports, '__esModule', { value: true })

module.exports.hello = function () {
  console.log('hello')
}
```

```ts
// index.ts
//@ts-expect-error - This is a test
import code from './lib'
code.hello()
```

```sh
bun build index.ts --outfile code.js # Same behavior with --compile
bun run code
```

### What is the expected behavior?

To print out `hello` in the terminal.

### What do you see instead?

```js
28 | // lib.js
29 | var require_lib = __commonJS((exports, module) => {
30 |   __export(exports, {
31 |     hello: () => {
32 |       {
33 |         return $hello;
                   ^
ReferenceError: Can't find variable: $hello
      at hello (/folder/code.js:33:15)
      at /folder/code.js:45:0
```

Generated JS file for reference:
```js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getProtoOf = Object.getPrototypeOf;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __toESM = (mod, isNodeMode, target) => {
  target = mod != null ? __create(__getProtoOf(mod)) : {};
  const to = isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target;
  for (let key of __getOwnPropNames(mod))
    if (!__hasOwnProp.call(to, key))
      __defProp(to, key, {
        get: () => mod[key],
        enumerable: true
      });
  return to;
};
var __commonJS = (cb, mod) => () => (mod || cb((mod = { exports: {} }).exports, mod), mod.exports);
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, {
      get: all[name],
      enumerable: true,
      configurable: true,
      set: (newValue) => all[name] = () => newValue
    });
};

// lib.js
var require_lib = __commonJS((exports, module) => {
  __export(exports, {
    hello: () => {
      {
        return $hello;
      }
    }
  });
  Object.defineProperty(exports, "__esModule", { value: true });
  exports.hello = function() {
    console.log("hello");
  };
});

// test.ts
var lib = __toESM(require_lib(), 1);
lib.default.hello();
```

### Additional information

If more reproducible examples is needed: https://github.com/elysiajs/elysia-html/issues/14.

Sorry if this issue has already been reported. I Did not find anything similar.

**Comment by Cubxx** [comment]
maybe it's solution: https://github.com/oven-sh/bun/issues/12463#issuecomment-2285353340

**Comment by Jarred-Sumner** [comment]
Fixed in Bun v1.1.22 by @paperdave
